core/src/impl/Kokkos_Core.cpp

```diff
@@ -608,8 +602,9 @@ void initialize(int& narg, char* arg[])
     }
     char * env_disablewarnings_str = std::getenv("KOKKOS_DISABLE_WARNINGS");
     if (env_disablewarnings_str!=nullptr) {
-        Impl::convert_uppercase(env_disablewarnings_str);
-        if ((strcmp(env_disablewarnings_str,"TRUE") == 0) || (strcmp(env_disablewarnings_str,"ON") == 0) || (strcmp(env_disablewarnings_str,"1") == 0))
+        std::string env_str (env_disablewarnings_str); // deep-copies string
+        for (char& c : env_str) { c = toupper (c); }
+        if ((env_str == "TRUE") || (env_str == "ON") || (env_str == "1"))
```

Comment:
Just for future information: You don't need the extra inner parentheses here, since `||` has lower precedence than `==`.